web/src/components/countryhistorycarbongraph.jsx

```diff
@@ -10,7 +10,8 @@ import AreaGraph from './graph/areagraph';
 import { getCO2IntensityByMode } from '../helpers/zonedata';
 
 const prepareGraphData = (historyData, co2ColorScale, electricityMixMode) => {
-  if (!historyData || !historyData[0]) {
+  if (!historyData || !historyData[0] || !historyData[0].stateDatetime) {
+    // Incomplete data
```

Comment:
This was added because missing `stateDatetime` caused some rendering errors. Turns out it's due to the initial data (before data is loaded from the backend)

Reply:
Yeah there are some odd states in between full data. This is where typescript would be very helpful

I'm changing to check for the .hasData property instead :)